controllers/openstackdataplanerole_controller.go

```diff
@@ -85,11 +86,78 @@ func (r *OpenStackDataPlaneRoleReconciler) SetupWithManager(mgr ctrl.Manager) er
 
 // ReconcileNodes ensure the desired state for the nodes
 func (r *OpenStackDataPlaneRoleReconciler) ReconcileNodes(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneRole) error {
-	// loop over r.Spec.DataPlaneNodes:
+	// loop over instance.Spec.DataPlaneNodes:
 	//   for each node:
 	//     (1) complete node.Spec based r.Spec.nodeTemplate, and values set on the
 	//         node (values on the node take precedence over those from the template)
 	//     (2) Create a CR of OpenStackDataPlaneNode from the node
 
+	for i := range instance.Spec.DataPlaneNodes {
```

Comment:
Do you need the index here or is it better using the current item returned by the iterator? 

e.g.

```
for i, cur := range instance.Spec.DataPlaneNodes {
 /***
  * e.g. node.Spec.Node.AnsibleHost = cur.Node.AnsibleHost
  ***/ 
}
```

This would help to make L114-L144 more robust.
I see you need the index on L100, I'm wondering if the index is required for node Name generation or we should have a diff solution.

Reply:
OK, I made that change.